src/components/designSystem/VerticalMenu.tsx

```diff
@@ -48,34 +50,58 @@ export const VerticalMenu = ({
       {!loading && tabs.length > 1 && (
         <div className="flex w-full flex-1 flex-col gap-1 overflow-visible">
           {tabs.map((tab, i) => {
-            const { link, hidden, title, beta, external, ...tabProps } = tab
+            const { link, hidden, title, beta, external, onAction, ...tabProps } = tab
 
             if (hidden) return null
 
-            return (
-              <ButtonLink
-                external={external}
-                title={title}
-                key={`${i}-${name}-${link}`}
-                to={link}
-                type="tab"
-                active={link === activeTab?.link}
-                onClick={!!onClick ? () => onClick(tab) : undefined}
-                {..._omit(tabProps, ['component', 'match'])}
-              >
-                <div className="flex w-full flex-row items-center justify-between">
-                  <div className="flex items-baseline gap-2">
-                    <Typography variant="body" color="inherit" noWrap>
-                      {title}
-                    </Typography>
-                    {!!beta && (
-                      <Chip size="small" label={translate('text_65d8d71a640c5400917f8a13')} />
-                    )}
+            if (link) {
+              return (
+                <ButtonLink
+                  external={external}
+                  title={title}
+                  key={`${i}-${name}-${link}`}
+                  to={link}
+                  type="tab"
+                  active={link === activeTab?.link}
+                  onClick={!!onClick ? () => onClick(tab) : undefined}
+                  {..._omit(tabProps, ['component', 'match'])}
+                >
+                  <div className="flex w-full flex-row items-center justify-between">
+                    <div className="flex items-baseline gap-2">
+                      <Typography variant="body" color="inherit" noWrap>
+                        {title}
+                      </Typography>
+                      {!!beta && (
+                        <Chip size="small" label={translate('text_65d8d71a640c5400917f8a13')} />
+                      )}
+                    </div>
+                    {!!external && <Icon name="outside" />}
                   </div>
-                  {!!external && <Icon name="outside" />}
-                </div>
-              </ButtonLink>
-            )
+                </ButtonLink>
+              )
+            }
+            if (onAction) {
```

Comment:
Should it be an `else if` here? The type for props does not explicitly prevent to pass both and I don't think it would be expected